### ISSUE TITLE
[Medium] Vulnerability: tough-cookie - Prototype Pollution

### ISSUE THREAD

**Issue by tom-vism** [nrwl/nx#20147]
### Current Behavior

Snyk has reported a medium vulnerability. The issue was found in nrwl/nx version 13.8.3 but it seems like even the latest version might have this issue (correct me if I'm wrong).

Introduced through @nrwl/angular@13.8.3 › @nrwl/jest@13.8.3 › jest-config@27.2.2 › jest-environment-jsdom@27.5.1 › jsdom@16.7.0 › tough-cookie@4.0.0
Note: it seems tough-cookie@4.1.2 version is used in master branch

Overview
_tough-cookie is a RFC6265 Cookies and CookieJar module for Node.js.
Affected versions of this package are vulnerable to Prototype Pollution due to improper handling of Cookies when using CookieJar in rejectPublicSuffixes=false mode. Due to an issue with the manner in which the objects are initialized, an attacker can expose or modify a limited amount of property information on those objects. There is no impact to availability._

### Expected Behavior

Fixed in tough-cookie@4.1.3
Snyk should not report issue

### Steps to Reproduce

Run snyk analysis

### Nx Report

```shell
Node : 18.18.2
   OS   : win32 x64
   npm  : 9.8.1

   nx : undefined
   @nrwl/angular : 13.8.3
   @nrwl/cli : 13.8.3
   @nrwl/cypress : 13.8.3
   @nrwl/detox : undefined
   @nrwl/devkit : 13.8.3
   @nrwl/eslint-plugin-nx : 13.10.6
   @nrwl/express : undefined
   @nrwl/jest : 13.8.3
   @nrwl/js : undefined
   @nrwl/linter : 13.8.3
   @nrwl/nest : undefined
   @nrwl/next : undefined
   @nrwl/node : undefined
   @nrwl/nx-cloud : undefined
   @nrwl/react : undefined
   @nrwl/react-native : undefined
   @nrwl/schematics : undefined
   @nrwl/storybook : 13.8.3
   @nrwl/tao : 13.8.3
   @nrwl/web : undefined
   @nrwl/workspace : 13.8.3
   typescript : 4.5.5
   rxjs : 6.6.7
   ---------------------------------------
   Community plugins:
         @angular/animations: 13.3.11
         @angular/cdk: 13.3.9
         @angular/common: 13.3.11
         @angular/compiler: 13.3.11
         @angular/core: 13.3.11
         @angular/forms: 13.3.11
         @angular/google-maps: 13.3.9
         @angular/platform-browser: 13.3.11
         @angular/platform-browser-dynamic: 13.3.11
         @angular/router: 13.3.11
         @fortawesome/angular-fontawesome: 0.10.2
         @angular-builders/jest: 13.0.4
         @angular-devkit/build-angular: 13.3.9
         @angular/cli: 13.3.8
         @angular/compiler-cli: 13.3.11
         @angular/language-service: 13.3.11
         @ngneat/spectator: 10.0.1
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by FrozenPandaz** [comment]
https://github.com/advisories/GHSA-72xf-g2v4-qvf3

**Comment by leosvelperez** [comment]
Thanks for reporting this!

The `jsdom@16.7.0` (the version you're using according to the logs) has a dependency `"tough-cookie": "^4.0.0"`. That range allows version 4.1.3, where the vulnerability is addressed. Most likely, the `tough-cookie` package was last resolved when 4.1.2 was the latest version matching the range, and because it's resolved to that version in your lock file, you'll get it installed every time.

New workspaces or clean package installations (e.g. deleting the lock file) will get version 4.1.3 of that package. Could you try deleting your lock file and installing the packages again?
